docker/src/main/java/brooklyn/location/docker/DockerHostLocation.java

```diff
@@ -156,24 +159,33 @@ public DockerContainerLocation obtain(Map<?,?> flags) throws NoMachinesAvailable
             // Add the entity Dockerfile if configured
             String dockerfile = entity.getConfig(DockerAttributes.DOCKERFILE_URL);
             String imageId = entity.getConfig(DockerAttributes.DOCKER_IMAGE_ID);
-            String imageName = DockerUtils.imageName(entity, dockerfile, repository);
+
+            Optional<String> baseImage = Optional.fromNullable(entity.getConfig(DockerAttributes.DOCKER_IMAGE_NAME));
+            String imageTag = Optional.fromNullable(entity.getConfig(DockerAttributes.DOCKER_IMAGE_TAG)).or("latest");
+            // TODO incorporate more info
+            final String imageName = DockerUtils.imageName(entity, dockerfile, repository);
+            final String fullyQualifiedImageName = Os.mergePaths(repository, imageName);
 
             // Lookup image ID or build new image from Dockerfile
-            LOG.warn("ImageName for entity {}: {}", entity, imageName);
-            String imageList = dockerHost.runDockerCommand("images --no-trunc " + Os.mergePaths(repository, imageName));
-            if (Strings.containsLiteral(imageList, imageName)) {
-                // Wait until committed before continuing
+            LOG.info("ImageName for entity {}: {}", entity, imageName);
+
+            if (dockerHost.getImageNamed(fullyQualifiedImageName, imageTag).isPresent()) {
```

Comment:
We should save the `Optional` here since we repeat this method call just to call `get()` on it a couple of lines later